packages/next/src/lib/typescript/writeConfigurationDefaults.ts

```diff
@@ -58,6 +58,7 @@ function getDesiredCompilerOptions(
     moduleResolution: {
       // In TypeScript 5.0, `NodeJs` has renamed to `Node10`
       parsedValue:
+        ts.ModuleResolutionKind.NodeNext ??
```

Comment:
I think we should use `bundler` instead of `nodenext` because `nodenext` requires file extensions just like Node.js does when using ESM. But `bundler` doesn't and its specifically designed for webpack/turbopack usage.

https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#moduleresolution-bundler